pkg/usagetracker/tracker.go

```diff
@@ -77,70 +98,73 @@ func (c *Config) Validate() error {
 	if c.IdleTimeout <= 0 || c.IdleTimeout > time.Hour {
 		return fmt.Errorf("invalid usage-tracker idle timeout %q, should be greater than 0 and less than 1 hour", c.IdleTimeout)
 	}
+	if c.EventsStorage.Writer.AutoCreateTopicEnabled && c.EventsStorage.Writer.AutoCreateTopicDefaultPartitions < c.Partitions {
+		return fmt.Errorf("number of configured partitions %d must be less or equal than the default number of partitions to be auto created %d for the Kafka topic", c.Partitions, c.EventsStorage.Writer.AutoCreateTopicDefaultPartitions)
+	}
 
 	return nil
 }
 
 type UsageTracker struct {
 	services.Service
 
-	store *trackerStore
+	instanceID int32
 
 	cfg        Config
 	bucket     objstore.InstrumentedBucket
 	overrides  *validation.Overrides
 	logger     log.Logger
 	registerer prometheus.Registerer
 
-	partitionID int32
-
 	// Partition and instance ring.
-	partitionLifecycler *ring.PartitionInstanceLifecycler
-	partitionRing       *ring.PartitionInstanceRing
-	instanceLifecycler  *ring.BasicLifecycler
+	partitionKVClient  kv.Client
+	instanceRing       *ring.Ring
+	partitionRing      *ring.PartitionInstanceRing
+	instanceLifecycler *ring.BasicLifecycler
 
 	// Events storage (Kafka).
 	eventsKafkaWriter *kgo.Client
-	eventsKafkaReader *kgo.Client
 
-	pendingCreatedSeriesMarshaledEvents chan []byte
+	mtx        sync.RWMutex
```

Comment:
[nit] I would rename it to `partitionsMtx` since it just protects `partitions`. We don't need to protect `lostPartitions` cause it's just used within `reconcilePartitions()` that is not called concurrently.

Reply:
Done.